### PR TITLE
feat(connector): [Cybersource] Use connector_response_reference_id as reference to merchant 

### LINKED ISSUE #2327: [FEATURE]: [Cybersource] Use `connector_response_reference_id` as reference to merchant
### :memo: Feature Description

- Reference id are used to map transactions in the connector’s dashboard.
- Hyperswitch manages several reference ids, such as `payment_id`, `attempt_id`, and `connector_transaction_id` for a single transaction. 
- However, merchants may encounter uncertainty when determining which ID to utilize in the connector dashboard to identify the payment.

### :hammer: Possible Implementation

- When we receive a response from the connector for the payment, we deserialize it and populate the `response` field in RouterData. 
- For the `TransactionResponse` type, we must fill the `connector_response_reference_id` with a corresponding reference id for the merchant to identify the transaction. 
- One might need to have exposure to api docs of the connector for which it is being implemented to decide what to fill in connector_response_reference_id.
- You can check this PR for further reference https://github.com/juspay/hyperswitch/pull/1735

🔖 Note: All the changes needed should be contained within `hyperswitch/crates/router/src/connector/`

### :package: Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### :package: Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by rashmibharambe**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Use connector_response_reference_id as reference to merchant for Cybersource
Fixes #2327 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
fixes #2327 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by SanchithHegde** [comment]
@rashmibharambe Could you merge the `main` branch into your PR?

**Comment by rashmibharambe** [comment]
> @rashmibharambe Could you merge the `main` branch into your PR?

Done, please review.

File: crates/router/src/connector/cybersource/transformers.rs [hunk 1]
```diff
@@ -313,12 +320,18 @@ impl<F, T>
...
                     redirection_data: None,
                     mandate_reference: None,
                     connector_metadata: None,
                     network_txn_id: None,
-                    connector_response_reference_id: None,
+                    connector_response_reference_id: item
+                        .response
+                        .client_reference_information
+                        .map(|cref| cref.code)
+                        .unwrap_or(Some(item.response.id.clone())),
```
Comment (srujanchikke) [inline crates/router/src/connector/cybersource/transformers.rs:334]: `clone` is not required here

File: crates/router/src/connector/cybersource/transformers.rs [hunk 2]
```diff
@@ -378,12 +392,16 @@ impl<F, T>
...
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: item
+                    .response
+                    .client_reference_information
+                    .map(|cref| cref.code)
+                    .unwrap_or(Some(item.response.id.clone())),
```
Comment (srujanchikke) [inline crates/router/src/connector/cybersource/transformers.rs:404]: `clone` is not required here .

Comment (rashmibharambe) [reply crates/router/src/connector/cybersource/transformers.rs]: Removed.

**Comment by rashmibharambe** [comment]
@srujanchikke - please review, comments addressed..

**Review by srujanchikke — APPROVED** [review]
Looks good to me!

**Review by ArjunKarthik — APPROVED** [review]

**Comment by ArjunKarthik** [comment]
@rashmibharambe Thanks for the PR

**Comment by VedantKhairnar** [comment]
Hey @rashmibharambe  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**